File: evita/dom/visuals/shape.js
// Copyright (c) 2016 Project Vogue. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

class Shape extends Node {
  /**
   * @param {!Document} document
   * @param {!NodeHandle} handle
   * @param {!Uint8Array} data
   */
  constructor(document, handle, data) {
    super(document, handle);
    this.data_ = data;
  }

  /** @return {!Uint8Array} */
  get data() { return this.data_; }

  /** @param {!Uint8Array} newData */
  set data(newData) {
    if (this.data_ === newData)
      return;
    this.data_ = newData;
    NodeHandle.setShapeData(this.handle_, newData);
  }

  /** @override @return {string} */
  get nodeName() { return '#shape'; }
}

File: evita/visuals/dom/shape.cc
// Copyright (c) 2015 Project Vogue. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "evita/visuals/dom/shape.h"

namespace visuals {

//////////////////////////////////////////////////////////////////////
//
// Shape
//
Shape::Shape(Document* document, const ShapeData& data)
    : Node(document, L"#shape"), data_(data) {}

Shape::~Shape() {}

}  // namespace visuals
